### DISCUSSION TITLE
Extension Development

### DISCUSSION THREAD

**Discussion by LunarSubterrane** [dotnet/interactive#3362] [Q&A] [answered]
I am trying to build an extension following the guide [here](https://github.com/dotnet/interactive/blob/main/docs/extending-dotnet-interactive.md) 

However, the microsoft.dotnet.interactive package references for that project seem to reference an older version of `ConnectKernelCommand` which does not support returning multiple Kernels.

The package microsoft.dotnet-interactive does not seem to be suited for building an extension project as the nuget restore fails due to "dotnettoolreference project style can only contain references of the dotnettool type".

What is the correct approach to referencing the right library to build the clock extension?

**Comment by jonsequitur** [comment, marked as answer]
We've just made some updates to the extension sample along with releasing the .NET 8 version of .NET Interactive. Please give it another try and let us know if it's working better for you.

**Comment by LunarSubterrane** [comment]
This fixed it. I pulled the latest nuget packes for microsoft.dotnet.interactive and microsoft.dotnet.interactive.csharp and it's now compiling as expected.
